packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -2081,79 +2079,48 @@ class _RenderAppBarTitleBox extends RenderAligningShiftedBox {
     final double textScaleFactor = math.min(MediaQuery.textScaleFactorOf(context), _kMaxTitleTextScaleFactor); // TODO(tahatesser): Add link to Material spec when available, https://github.com/flutter/flutter/issues/58769.
     final FlexibleSpaceBarSettings settings = context.dependOnInheritedWidgetOfExactType<FlexibleSpaceBarSettings>()!;
-    final double topPadding = primary ? MediaQuery.viewPaddingOf(context).top : 0;
-    final double collapsedHeight = settings.minExtent - topPadding - bottomHeight;
-    final double scrollUnderHeight = settings.maxExtent - (settings.minExtent / textScaleFactor) + bottomHeight;
-    final _ScrollUnderFlexibleConfig config;
-    switch (variant) {
-      case _ScrollUnderFlexibleVariant.medium:
-        config = _MediumScrollUnderFlexibleConfig(context);
-      case _ScrollUnderFlexibleVariant.large:
-        config = _LargeScrollUnderFlexibleConfig(context);
-    }
+    final _ScrollUnderFlexibleConfig config = configBuilder(context);
+    assert(
+      config.expandedTitlePadding.isNonNegative,
+      'The _ExpandedTitleWithPadding widget assumes that the expanded title padding is non-negative. '
+      'Update its implementation to handle negative padding.',
+    );
 
-    late final Widget? expandedTitle;
-    if (title != null) {
-      final TextStyle? expandedTextStyle = titleTextStyle
-        ?? appBarTheme.titleTextStyle
-        ?? config.expandedTextStyle?.copyWith(
-             color: foregroundColor ?? appBarTheme.foregroundColor ?? defaults.foregroundColor,
-           );
-      expandedTitle = config.expandedTextStyle != null
-        ? DefaultTextStyle(
-            style: expandedTextStyle!,
-            child: title!,
-          )
-        : title;
-    }
+    final TextStyle? expandedTextStyle = titleTextStyle
+      ?? appBarTheme.titleTextStyle
+      ?? config.expandedTextStyle?.copyWith(color: foregroundColor ?? appBarTheme.foregroundColor ?? defaults.foregroundColor);
 
-    EdgeInsetsGeometry expandedTitlePadding() {
-      final EdgeInsets padding = config.expandedTitlePadding!.resolve(Directionality.of(context));
-      if (bottomHeight > 0) {
-        return EdgeInsets.fromLTRB(padding.left, 0, padding.right, bottomHeight);
-      }
-      if (theme.useMaterial3) {
-        final TextStyle textStyle = config.expandedTextStyle!;
-        // Substract the bottom line height from the bottom padding.
-        // TODO(tahatesser): Figure out why this is done.
-        // https://github.com/flutter/flutter/issues/120672
-        final double adjustBottomPadding = padding.bottom
-          - (textStyle.fontSize! * textStyle.height! - textStyle.fontSize!) / 2;
-        return EdgeInsets.fromLTRB(
-          padding.left,
-          0,
-          padding.right,
-          adjustBottomPadding / textScaleFactor,
-        );
-      }
-      return padding;
-    }
+    final Widget? expandedTitle = switch ((title, expandedTextStyle)) {
```

Comment:
This is slick but I'm not sure the logic is the same or even simpler than before.
```dart
    if (title != null) {
      ...
      expandedTitle = config.expandedTextStyle != null
        ? DefaultTextStyle(
            style: expandedTextStyle!,
            child: title!,
          )
        : title;
    }
 ```

Would only wrap the title with the DefaultTextStyle if `config.expandedTextStyle` was non-null. The new code checks `expandedTextStyle` which incorporates the widget/theme defaults.  I don't know why the code was the way it was, so I suppose if the change doesn't break anything it's OK.

Reply:
It's from https://github.com/flutter/flutter/pull/122542/files#diff-20c19d254b0a82027625f1246ba51cb2264bfabf105fe91d532c19e6a6d77478L2141-L2151

@TahaTesser does this change look ok to you?

Reply:
This looks good to me! 

`config.expandedTextStyle != null` existed before I added `title ?? appBarTheme.titleTextStyle`. I should have updated it to just `expandedTextStyle != null`.  Yours refactor makes it much cleaner and hard to miss.